src/TRoom.cpp

```diff
@@ -703,95 +725,630 @@ void TRoom::restore( QDataStream & ifs, int roomID, int version )
+        // A real exit - should have a real destination, and NOT have a stub
+        if( Q_UNLIKELY( ! mpRoomDB->getRoom( exitRoomId ) ) ) {
+            // But it doesn't exist
+            QString auditKey = QStringLiteral( "audit.made_stub_of_valid_but_missing_exit.%1" ).arg( dirCode );
+            QString warnMsg = tr( "[ WARN ]  - Room with Id: %1 has an exit \"%2\" with an exit to: %3 but that room\n"
+                                              "does not exist.  The exit will be removed (but the destination room\n"
+                                              "Id will be stored in the room user data under a key:\n"
+                                              "\"%4\")\n"
+                                              "and the exit will be turned into a stub." )
+                                  .arg( id )
+                                  .arg( displayName )
+                                  .arg( exitRoomId )
+                                  .arg( auditKey );
+            mpRoomDB->mpMap->postMessage( warnMsg );
+            userData.insert( auditKey, QString::number( exitRoomId ) );
+            if( ! exitStubs.contains( dirCode ) ) {
+                exitStubs.append( dirCode ); // Add a stub (this is so we can retain doors, though exit weights, custom lines and locks will go)
+                exitStubsPool.remove( dirCode ); // Remove a (now valid) stub in this direction from check pool
+            }
+
+            exitRoomId = -1;
+
+            // Remove the following things that a stub exit does not handle
+            // And eliminate the corresponding things from the pools of things
+            // that have to be checked:
+            // TODO: Add additional warnings if we ARE deleting any data in following
+            exitLocks.removeAll( dirCode );
+            exitLocksPool.remove( dirCode );
+
+            exitWeights.remove( doorAndWeight );
+            exitWeightsPool.remove( doorAndWeight );
+
+            doorsPool.remove( doorAndWeight ); // we now have a stub exit and that can take a door so clear the door from the check pool
+
+            customLines.remove( customLine );
+            customLinesPool.remove( customLine );
+
+            customLinesColor.remove( customLine );
+            customLinesColorPool.remove( customLine );
+
+            customLinesStyle.remove( customLine );
+            customLinesStylePool.remove( customLine );
+
+            customLinesArrow.remove( customLine );
+            customLinesArrowPool.remove( customLine );
+        }
+        else {
+            // we do have a valid exit destination room
+
+            // We cannot allow a stub exit at the same time as a real exit:
+            if( exitStubs.contains( dirCode ) ) {
+                QString warnMsg = tr( "[ ALERT ] - Room with Id: %1 has an exit \"%2\" with an exit to: %3 but also\n"
+                                                  "has a record of a stub exit!  That stub will be removed." )
```

Comment:
Does this mean "has a record of a stub exit [in the same direction]"? Because a stub exit on its own isn't a bad thing.